src/templates.ts

```diff
@@ -87,6 +110,27 @@ export const downloadTemplates =
     }
   }
 
+const enrichWithPrePendingHeader =
+  (mustacheRenderedContent: string, template: Template, codeowners: string) =>
+  (log: Logger): string => {
+    // TODO Check if file extensions fits together
+    const templateExtension = template.destinationPath.split('.').pop() as string
+    if (!(templateExtension == 'yaml' || templateExtension == 'toml' || templateExtension == 'yml')) {
+      log.debug(`File extension: ${templateExtension} with not supported comments`)
+      return mustacheRenderedContent
+    }
+
+    const codeOwnersEntries = parse(codeowners)
+    const matchedWithCodeOwner = matchFile(template.sourcePath, codeOwnersEntries)
+
+    if (!process.env.PREPENDING_HEADER_TEMPLATE) {
+      log.info("Prepending header template not defined, using default.")
+      return `#OWNER: ${matchedWithCodeOwner?.owners}\n\n` + mustacheRenderedContent
+    }
+
+    return `${process.env.PREPENDING_HEADER_TEMPLATE + matchedWithCodeOwner?.owners}\n\n` + mustacheRenderedContent
```

Comment:
These can be inlined into one string in a single pass 😄 

```suggestion
    const header = process.env.PREPENDING_HEADER_TEMPLATE
    if (!header) {
      log.info("Prepending header template not defined, using default.")
      return `#OWNER: ${matchedWithCodeOwner?.owners}\n\n${mustacheRenderedContent}`
    }

    return `${header}${matchedWithCodeOwner?.owners}\n\n${mustacheRenderedContent}`
```

Reply:
This basically says "try to fetch the header, if it's not there, use the default". That way a header is always present and we can just return a single string 👍🏼

Reply:
I see your point, however, have some issues with `process.env.PREPENDING_HEADER_TEMPLATE` as it resolved to an empty string which is "true" for both conditions. 🤔 Setting it up to undefined in tests leads to an "undefined" string 😅

Reply:
With [commit](https://github.com/pleo-oss/file-distributor/pull/39/commits/a183252431cc0ba5ceb0d1c7231061a1eb191aa5) applied your suggestion but needed to stay with `?` structure instead of `??` to pass tests.

Reply:
Ahhh, it's because it's the empty string, probably. Makes sense!